File: isma-hsm/src/main/java/ru/nstu/isma/core/hsm/linear/HMLinearEquation.java
package ru.nstu.isma.core.hsm.linear;

import ru.nstu.isma.core.hsm.exp.HMExpression;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Created by Bessonov Alex
 * on 13.03.2015.
 */
public class HMLinearEquation implements Serializable {
    private ArrayList<HMExpression> leftPart;
    private HMExpression rightPart;
    private Integer size;

    public HMLinearEquation(Integer size) {
        this.size = size;
        leftPart = new ArrayList<>(size);
    }

    public HMExpression getRightPart() {
        return rightPart;
    }

    public void setRightPart(HMExpression rightPart) {
        this.rightPart = rightPart;
    }

    public void setEquationElem(HMLinearVar var, HMExpression expression) {
        leftPart.add(var.getColumnIndex(), expression);
    }

    public ArrayList<HMExpression> getLeftPart() {
        return leftPart;
    }
}

File: isma-next-hsm/src/main/kotlin/ru/nstu/isma/hsm/linear/HMLinearEquation.kt
package ru.nstu.isma.hsm.linear

import ru.nstu.isma.hsm.exp.HMExpression
import java.io.Serializable
import java.util.*

/**
 * Created by Bessonov Alex
 * on 13.03.2015.
 */
class HMLinearEquation(private val size: Int) : Serializable {
    val leftPart: ArrayList<HMExpression?> = ArrayList(size)
    var rightPart: HMExpression? = null
    fun setEquationElem(`var`: HMLinearVar, expression: HMExpression?) {
        leftPart.add(`var`.columnIndex!!, expression)
    }

}